### PR TITLE
fix: deprecate `window.web3` and use `@metamask/detect-provider` for provider detection

### LINKED ISSUE #288: Remove window.web3 API

### PR DESCRIPTION
**Pull request by evgenyboxer**
Fixes #288

### CODE REVIEW

File: utils/network.ts [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
-		if (window?.web3?.eth?.net) {
-			const networkId = await window.web3.eth.net.getId();
-			return Number(networkId);
-		} else if (window?.web3?.version?.network) {
-			return Number(window?.web3.version.network);
-		} else if (window?.ethereum?.networkVersion) {
-			return Number(window?.ethereum?.networkVersion);
+		const provider = (await detectEthereumProvider()) as EthereumProvider;
+		if (provider) {
+			return Number(provider.networkVersion);
```
Comment (andytcf) [inline utils/network.ts:16]: ```return provider ? Number(provider.networkVersion) : DEFAULT_NETWORK_ID; ``` @evgenyboxer

**Review by andytcf — CHANGES_REQUESTED** [review]
LGTM, left a comment if you want to refactor or not

File: utils/network.ts (hunk 1 above)
Comment (evgenyboxer) [reply utils/network.ts]: thanks! pushed a fix now.

**Comment by evgenyboxer** [comment]
@andytcf ready for review

**Review by clementbalestrat — APPROVED** [review]

**Pull request merged**